pkg/auth/auth_provider.go

```diff
@@ -0,0 +1,26 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package auth
+
+import "net/http"
+
+// Provider provide a general method to add auth message
+type Provider interface {
```

Comment:
Sure. I will update it.

Reply:
Ah. It is under the `auth` package. Because there has a package name represents the `Provider` in which package.  When we use it we will call `auth.Provider`. If we call it `AuthProvider`, it will duplicate with the package name.

Reply:
It does not recommend that the method name starts with the package name in golang.

Reply:
ok